### PR TITLE
Bugfixes in write.magpie and write.report

### PR DESCRIPTION
**Pull request by tscheypidi**
- fixed a bug in write.magpie in which the comment character was missing at the beginning of comments in csv files if line breaks were being used in the comment

- fixed another bug in which magclass:::unitsplit was working incorrectly for R version < 4.0 (which had a different default setting for stringsAsFactors in various functions)

### CODE REVIEW

File: .github/workflows/lucode2-check.yaml [hunk 1]
```diff
@@ -56,6 +56,18 @@ jobs:
 
       - uses: r-lib/actions/setup-pandoc@v1
 
+      - name: Cache R libraries
+        uses: actions/cache@v3
```
Comment (mikapfl) [inline .github/workflows/lucode2-check.yaml:60]: you used an old lucode2 version to generate this. :wink:

File: R/write.magpie.R [hunk 2]
```diff
@@ -119,8 +119,9 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
     }
 
     # look for comment/additional information
-    if (is.null(comment) & !is.null(attr(x, "comment"))) comment <- attr(x, "comment")
+    if (is.null(comment) && !is.null(attr(x, "comment"))) comment <- attr(x, "comment")
```
Comment (mikapfl) [inline R/write.magpie.R:122]: what is the effect of the additional `&`?

**Review by mikapfl — APPROVED** [review]
Looks fine to me, I just don't know what the additional binary operator does everywhere.

File: R/write.magpie.R (hunk 2 above)
Comment (tscheypidi) [reply R/write.magpie.R:122]: As far as I understand `&` can do vector comparisons and you might end up with a vector of TRUE and FALSE statements. `&&` can only compare single logical values but has the advantage that the second statement is not being evaluated if the first statement is already FALSE. I changed it in the code as my lintr version was complaining about and the `&&` and `||` is indeed the better version here while it probably does not make a big difference here what to take

**Pull request merged**